### ISSUE TITLE
Deep Merge non arrays

### ISSUE THREAD

**Issue by stevendelarwelle** [danielsdeleo/deep_merge#44]
Is it possible to deep merge this
{:available=>{:inventory=>{:sordernum=>849836}}}
{:available=>{:inventory=>{:sordernum=>221836}}}

to produce 
{:available=>{:inventory=>[{:sordernum=>849836},{:sordernum=>221836}]}}

**Comment by Fryguy** [comment]
No, there's no option for that from what I can tell.  We'd need a new option, so this is an enhancement request.  That being said, I'm not sure exactly how it would know when to make something into an Array or not.
